libkbfs/disk_limits_unix.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+// +build !windows
+
+package libkbfs
+
+import (
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// getDiskLimits gets the disk limits for the logical disk containing
+// the given path.
+//
+// TODO: Also return available files.
+func getDiskLimits(path string) (availableBytes uint64, err error) {
+	var stat unix.Statfs_t
+	err = unix.Statfs(path, &stat)
+	if err != nil {
+		return 0, errors.WithStack(err)
+	}
+
+	// Bavail is the free block count for an unprivileged user.
+	availableBytes = stat.Bavail * uint64(stat.Bsize)
+
+	// TODO: Use stat.Ffree to return availableFiles.
+
+	return availableBytes, nil
```

Comment:
I'm conflicted about whether the disk limit should be based on available bytes, rather than total bytes.  The argument against using available bytes is that it will change each time we restart KBFS, and the available bytes will be reduced by anything that's already in the journal on restart!  So I find it hard to reason about in the long run.  Of course using total bytes has the issue that the disk could already be full, but I'm inclined to say there's not much we can do about that, since the disk could fill up anyway after we start and we have to handle it with a nice error somehow anyway.  What do you think?